spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/leader/LeaderInitiator.java

```diff
@@ -263,30 +299,36 @@ protected class LeaderSelector implements Callable<Void> {
 
 		protected final String role = LeaderInitiator.this.candidate.getRole();
 
-		private volatile boolean locked = false;
+		private volatile boolean leader = false;
 
 		@Override
 		public Void call() {
 			try {
 				while (isRunning()) {
 					try {
-						if (LeaderInitiator.this.lock.isLocked()) {
+						if (logger.isDebugEnabled()) {
+							logger.debug("Am I the leader (" + LeaderInitiator.this.candidate.getRole() + ") ? "
+									+ this.leader);
+						}
+						if (LeaderInitiator.this.getLock().isLockedByCurrentThread()) {
+							if (!this.leader) {
+								// Since we have the lock we need to ensure that the leader flag is set
+								this.leader = true;
+							}
 							// Give it a chance to expire.
 							if (yieldSign.tryAcquire(LeaderInitiator.this.heartBeatMillis, TimeUnit.MILLISECONDS)) {
 								revokeLeadership();
 								// Give it a chance to elect some other leader.
 								Thread.sleep(LeaderInitiator.this.busyWaitMillis);
 							}
-						}
-						else {
+						} else {
```

Comment:
I will fix

Reply:
Just for the future contribution: now need to say me if you agree with the change.
I will see that in your subsequent commits anyway.
Otherwise such an `Agreed` causes a noise in my mail box.
And I have like tons of PRs and issue to review/react 😄